presto-common/src/main/java/com/facebook/presto/common/block/DictionaryBlock.java

```diff
@@ -264,14 +264,28 @@ public long getRegionLogicalSizeInBytes(int positionOffset, int length)
         }
 
         long sizeInBytes = 0;
-        long[] seenSizes = new long[dictionary.getPositionCount()];
-        Arrays.fill(seenSizes, -1L);
-        for (int i = positionOffset; i < positionOffset + length; i++) {
-            int position = getId(i);
-            if (seenSizes[position] < 0) {
-                seenSizes[position] = dictionary.getRegionLogicalSizeInBytes(position, 1);
+        // Dictionary Block may contain large number of keys and small number of indices.
+        // This happens during the file read or write, where the dictionary of the file
+        // is much larger than number of rows read. In these cases, compute the size
```

Comment:
Let's maybe have a generic comment. Presto main engine doesn't have the concept of files.

Reply:
Remove the reference to file